extern/agg24-svn/include/agg_image_filters.h

```diff
@@ -53,16 +53,15 @@ namespace agg
             double r = filter.radius();
             realloc_lut(r);
             unsigned i;
-            unsigned pivot = diameter() << (image_subpixel_shift - 1);
-            for(i = 0; i < pivot; i++)
+            unsigned pivot = (diameter() << (image_subpixel_shift - 1)) - 1;
+            for(i = 0; i <= pivot + 1; i++)
             {
                 double x = double(i) / double(image_subpixel_scale);
                 double y = filter.calc_weight(x);
-                m_weight_array[pivot + i] = 
-                m_weight_array[pivot - i] = (int16)iround(y * image_filter_scale);
+                int16 value = iround(y * image_filter_scale);
+                m_weight_array[pivot + i] = value;
+                if(i <= pivot) m_weight_array[pivot - i] = value;
             }
-            unsigned end = (diameter() << image_subpixel_shift) - 1;
-            m_weight_array[0] = m_weight_array[end];
```

Comment:
This line is one of the bugs.  `0` and `end` are different distances from `pivot`, so they should not be set equal to each other.

Reply:
It turns out that I can't add a preprocessor definition in `_image_resample.h` a la #28122.  There is buggy code in `agg_image_filters.cpp` that needs to be skipped, but since it is a CPP file rather than a H file, it gets compiled separately with no awareness of `_image_resample.h`.  I have put the preprocessor definition here at the top of `agg_image_filters.h`, which is lower level than would be nice, but at least still makes it clear what is custom code.

Reply:
You could also put the define in the corresponding `meson.build` if you prefer.